src/util-dpdk-rte-flow.c

```diff
@@ -0,0 +1,931 @@
+        if (flow_handler_info == NULL) {
+            goto bypass_fail;
+        }
+        flow_handler_info->flow = flow;
+        flow_handler_info->src_handler = src_handler;
+        flow_handler_info->dst_handler = dst_handler;
+        flow_handler_info->dpdk_vars = flow->livedev->dpdk_vars;
+        fc->bypass_data = flow_handler_info;
+        fc->BypassUpdate = RteBypassUpdate;
+        fc->BypassFree = RteBypassFree;
+        LiveDevAddBypassStats(flow->livedev, 1, family);
+        LiveDevAddBypassSuccess(flow->livedev, 1, family);
+        SCReturnInt(1);
+    }
+
+bypass_fail:;
+    struct rte_flow_error flow_error = { 0 };
+    int retval = rte_flow_destroy(flow->livedev->dpdk_vars->port_id, src_handler, &flow_error);
+    if (retval != 0) {
+        SCLogError("rte_flow dynamic bypass: destroy rte_flow rule error %s errmsg: %s",
+                rte_strerror(-retval), flow_error.message);
+    }
+
+    retval = rte_flow_destroy(flow->livedev->dpdk_vars->port_id, dst_handler, &flow_error);
+    if (retval != 0) {
+        SCLogError("rte_flow dynamic bypass: destroy rte_flow rule error %s errmsg: %s",
+                rte_strerror(-retval), flow_error.message);
+    }
+    LiveDevAddBypassFail(flow->livedev, 1, family);
+    SCReturnInt(0);
+}
+
+int RteFlowBypassCallback(Packet *p)
+{
+    if (p == NULL || p->flow == NULL) {
+        SCReturnInt(0);
+    }
+
+    /* Only bypass TCP and UDP */
+    if (!(PacketIsTCP(p) || PacketIsUDP(p))) {
+        SCReturnInt(0);
+    }
+
+    FlowKey *flow_key = NULL;
+    RteFlowBypassData *rte_flow_bypass_data = p->livedev->dpdk_vars->rte_flow_bypass_data;
+    if (rte_mempool_get(rte_flow_bypass_data->bypass_mp, (void **)&flow_key) < 0) {
+        SCLogError("Memory allocation for rte_flow bypass data failed");
+        SCReturnInt(0);
+    }
+
+    if (PacketIsIPv4(p)) {
+        flow_key->src.family = AF_INET;
+        flow_key->src.address.address_un_data32[0] = (GET_IPV4_SRC_ADDR_U32(p));
+        flow_key->dst.family = AF_INET;
+        flow_key->dst.address.address_un_data32[0] = (GET_IPV4_DST_ADDR_U32(p));
+    } else if (PacketIsIPv6(p)) {
+        flow_key->src.family = AF_INET6;
+        memcpy(flow_key->src.address.address_un_data8, GET_IPV6_SRC_ADDR(p), 16 * sizeof(uint8_t));
+        flow_key->dst.family = AF_INET6;
+        memcpy(flow_key->dst.address.address_un_data8, GET_IPV6_DST_ADDR(p), 16 * sizeof(uint8_t));
+    }
+    if (p->proto == IPPROTO_TCP) {
+        flow_key->proto = IPPROTO_TCP;
+    } else {
+        flow_key->proto = IPPROTO_UDP;
+    }
+    flow_key->sp = p->sp;
+    flow_key->dp = p->dp;
+    flow_key->livedev_id = p->livedev->id;
+    flow_key->vlan_id[0] = p->vlan_id[0];
+    flow_key->vlan_id[1] = p->vlan_id[1];
+    flow_key->vlan_id[2] = p->vlan_id[2];
+    flow_key->recursion_level = 0;
+
+    int retval = rte_ring_mp_enqueue(rte_flow_bypass_data->bypass_ring, flow_key);
+    /* If ring is full, continue with local bypass */
+    if (retval < 0) {
```

Comment:
Important: I would appreciate stats such as the number of failed dequeues, failed rte_flow creations, failed stats queries, etc.